Assets/Scripts/Gameplay/GameManager.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Tilemaps;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+using System;
+using System.Linq;
+
+//It's a singleton so call this using GameManager.Instance.
+public class GameManager : Singleton<GameManager>
+{
+    public GameObject losePanel, winPanel;
+    
+    public GameObject[] playersInTeamRed, playersInTeamBlue, itemsToBeSpawned;
+    private int goalsScoredRed, goalsScoredBlue;
+    public int goalsToWin = 3, itemSpawnDelay = 30;
```

Comment:
Should `itemSpawnDelay` be an integer? It feels like we may manipulate this value with other values in the future, consider converting to a float.